File: Csharp_class/InterestEarningAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*红利帐户：
 * 将获得月末余额 2% 的额度
 
 */
namespace Csharp_class
{
    public class InterestEarningAccount:BankAccount
    {

        public InterestEarningAccount(string name, decimal initialBalance) : base(name, initialBalance)
        {

        }
        public override void PerformMonthEndTransactions()
        {
            if (Balance > 500m)
            {
                var interest = Balance * 0.05m;
                MakeDeposit(interest, DateTime.Now, "按月申请利息");
            }
        }
    }
}

File: Csharp_class/cpp_bank_system/src/InterestEarningAccount.cpp
#include "../include/InterestEarningAccount.h"

// 构造函数实现
InterestEarningAccount::InterestEarningAccount(const std::string& name, double initialBalance)
    : BankAccount(name, initialBalance) {
}

// 重写月末交易处理方法
void InterestEarningAccount::performMonthEndTransactions() {
    if (getBalance() > 500.0) {
        double interest = getBalance() * 0.05;  // 5%利息（对应C#代码中的0.05m）
        makeDeposit(interest, std::chrono::system_clock::now(), "按月申请利息");
    }
}